IDE/eslint doesn't report un-imported `$` as undefined any more

Maybe caused by a related dependency change by Update JS and PY dependencies (#32482) and/or some recent upgrades.

It seems that the eslint dependency is also broken.

cc @silverwind 

```
# comment out the first `import $ from 'jquery';` line
#
# the commit before #32482
$ git checkout cad313e64f01146b34b66ed6aa1344d9002dfe6c
$ npm i && npx eslint web_src/js/features/repo-issue-content.ts 
/Users/xiaoguang/work/gitea/web_src/js/features/repo-issue-content.ts
   14:17  error  '$' is not defined  no-undef

$ git checkout main
$ npm i && npx eslint web_src/js/features/repo-issue-content.ts
(no eslint error)
(but there are other npm warnings)

```
<details>

Before:

![image](https://github.com/user-attachments/assets/f22645aa-e701-439b-bd23-cde14e32a4e8)

After:

![image](https://github.com/user-attachments/assets/ad38330a-b89c-405d-9ceb-86853c936079)

</details>

--- a/.eslintrc.cjs
+++ b/.eslintrc.cjs
@@ -674,7 +674,7 @@ module.exports = {
     'no-this-before-super': [2],
     'no-throw-literal': [2],
     'no-undef-init': [2],
-    'no-undef': [0],
+    'no-undef': [2], // it is still needed by eslint & IDE to prompt undefined names in real time
     'no-undefined': [0],
     'no-underscore-dangle': [0],
     'no-unexpected-multiline': [2],

--- a/package.json
+++ b/package.json
@@ -86,7 +86,7 @@
     "eslint": "8.57.0",
     "eslint-import-resolver-typescript": "3.7.0",
     "eslint-plugin-array-func": "4.0.0",
-    "eslint-plugin-github": "5.1.4",
+    "eslint-plugin-github": "5.0.2",
     "eslint-plugin-import-x": "4.6.1",
     "eslint-plugin-no-jquery": "3.1.0",
     "eslint-plugin-no-use-extend-native": "0.5.0",

